src/artifactorypurge/artifactorypurge.py

```diff
@@ -1,64 +1,81 @@
+import os
+
 import click
-import imp
-from .utils.logging import getLogger
-from .credentials import load_credentials
+from pluginbase import PluginBase
 
+from .credentials import load_credentials
 from .utils.artifactory import Artifactory
+from .utils.logging import getLogger
 from .utils.performance import get_performance_report
 
 LOG = getLogger(__name__)
 
+
 @click.command()
-@click.option('--dryrun/--nodryrun', default=True, is_flag=True, help='Dryrun does not delete any artifacts. On by default')
-@click.option('--reponame', default=None, help='Operate on single repo')
-@click.option('--project', default=None, help='Operate on single project')
+@click.option(
+    '--dryrun/--nodryrun', default=True, is_flag=True, help='Dryrun does not delete any artifacts. On by default')
+@click.option('--default/--no-default', default=True, is_flag=True, help='If false, does not apply default policy')
+@click.option('--policies-path', required=False, help='Path to extra policies directory')
 #@click.argument('url')
-def purge(dryrun, reponame, project): #, url):
-    exceptions = {}
-
+def purge(dryrun, policies_path, default):  #, url):
     credentials = load_credentials()
-    artifactory = Artifactory(credentials['artifactory_url'], 
-                              credentials['artifactory_username'],
+    artifactory = Artifactory(credentials['artifactory_url'], credentials['artifactory_username'],
                               credentials['artifactory_password'])
 
-    before = artifactory.list(reponame)
+    plugin_source = setup_pluginbase(extra_policies_path=policies_path)
```

Comment:
I think you'd probably want to add a `LOG.debug` to show you all the plugins found. I'm sure we will need this later when troubleshooting.

Reply:
added